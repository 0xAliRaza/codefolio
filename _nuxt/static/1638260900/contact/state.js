window.__NUXT__=(function(a){return {staticAssetsBase:"\u002F_nuxt\u002Fstatic\u002F1638260900",layout:"default",error:a,serverRendered:true,routePath:"\u002Fcontact",config:{_app:{basePath:"\u002F",assetsPath:"\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"084e2058"}}}}(null));