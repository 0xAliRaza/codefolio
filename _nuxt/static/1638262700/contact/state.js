window.__NUXT__=(function(a){return {staticAssetsBase:"\u002F_nuxt\u002Fstatic\u002F1638262700",layout:"default",error:a,serverRendered:true,routePath:"\u002Fcontact",config:{_app:{basePath:"\u002F",assetsPath:"\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"67d6e611"}}}}(null));