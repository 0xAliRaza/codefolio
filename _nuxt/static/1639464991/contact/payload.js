__NUXT_JSONP__("/contact", (function(a,b,c,d,e,f){return {data:[{skills:{slug:"skills",skills:["HTML","JavaScript","CSS","SCSS","Bootstrap","Angular","Vue.js","Nuxt.js","PHP","Laravel","Bash","Docker","JQuery","Webpack"],dir:c,path:"\u002Fskills",extension:d,createdAt:e,updatedAt:e},navbar:{slug:"navbar",resumeUrl:a,externalLinks:[{text:f,url:"https:\u002F\u002Fgithub.com\u002Fuser"},{text:"blog",url:"https:\u002F\u002Fblog.user.com\u002F"}],dir:c,path:"\u002Fnavbar",extension:d,createdAt:b,updatedAt:b},contact:{slug:"contact",formAction:a,socialIcons:[{name:"discord",url:a},{name:"facebook",url:a},{name:f,url:a},{name:"twitter",url:a},{name:"gmail",url:a},{name:"skype",url:a}],dir:c,path:"\u002Fcontact",extension:d,createdAt:b,updatedAt:b}}],fetch:{},mutations:void 0}}("#","2021-12-14T06:55:39.948Z","\u002F",".json","2021-12-14T06:55:39.952Z","github")));