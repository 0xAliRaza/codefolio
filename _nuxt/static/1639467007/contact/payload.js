__NUXT_JSONP__("/contact", (function(a,b,c,d,e){return {data:[{skills:{slug:"skills",skills:["HTML","JavaScript","CSS","SCSS","Bootstrap","Angular","Vue.js","Nuxt.js","PHP","Laravel","Bash","Docker","JQuery","Webpack"],dir:c,path:"\u002Fskills",extension:d,createdAt:b,updatedAt:b},navbar:{slug:"navbar",resumeUrl:a,externalLinks:[{text:e,url:"https:\u002F\u002Fgithub.com\u002Fuser"},{text:"blog",url:"https:\u002F\u002Fblog.user.com\u002F"}],dir:c,path:"\u002Fnavbar",extension:d,createdAt:b,updatedAt:b},contact:{slug:"contact",formAction:a,socialIcons:[{name:"discord",url:a},{name:"facebook",url:a},{name:e,url:a},{name:"twitter",url:a},{name:"gmail",url:a},{name:"skype",url:a}],dir:c,path:"\u002Fcontact",extension:d,createdAt:b,updatedAt:b}}],fetch:{},mutations:void 0}}("#","2021-12-14T07:29:16.762Z","\u002F",".json","github")));