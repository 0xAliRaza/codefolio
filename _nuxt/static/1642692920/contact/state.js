window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1642692920",layout:"default",error:a,serverRendered:true,routePath:"\u002Fcontact",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"02c398a8"}}}}(null));