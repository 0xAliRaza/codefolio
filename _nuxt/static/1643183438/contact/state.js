window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1643183438",layout:"default",error:a,serverRendered:true,routePath:"\u002Fcontact",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"df474f23"}}}}(null));