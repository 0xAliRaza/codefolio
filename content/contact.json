{
  "formAction": "#",
  "location": "Example, USA.",
  "email": "contact@example.com",
  "phone": "+1-555-555-0100"
}
