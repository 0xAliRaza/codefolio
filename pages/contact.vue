<template>
  <div>
    <Header
      :resume-url="navbar.resumeUrl"
      :external-links="navbar.externalLinks"
    />
    <main class="main">
      <div class="container">
        <div class="row justify-content-center">
          <div class="col-12 col-md-10 col-lg-8 col-xl-7 col-xxl-6">
            <section class="contact">
              <div class="contact__heading text-center my-3">
                <h1>Contact</h1>
              </div>

              <!-- Add your custom action here (i.e. formspree.io) -->
              <form
                v-if="contact"
                :action="contact.formAction"
                method="POST"
                class="contact__form py-3"
              >
                <div class="mb-3">
                  <label for="fullname" class="form-label contact__label"
                    >Full name</label
                  >
                  <input
                    id="fullname"
                    type="text"
                    class="form-control contact__input"
                    name="fullname"
                    required
                  />
                </div>
                <div class="mb-3">
                  <label for="email" class="form-label contact__label"
                    >Email</label
                  >
                  <input
                    id="email"
                    type="email"
                    class="form-control contact__input"
                    name="email"
                    required
                  />
                </div>
                <div class="mb-3">
                  <label for="subject" class="form-label contact__label"
                    >Subject</label
                  >
                  <input
                    id="subject"
                    type="text"
                    class="form-control contact__input"
                    name="subject"
                    required
                  />
                </div>
                <div class="mb-3">
                  <label for="message" class="form-label contact__label"
                    >Message</label
                  >
                  <textarea
                    id="message"
                    class="form-control contact__textarea"
                    name="message"
                    rows="4"
                    required
                  ></textarea>
                </div>
                <div class="d-flex justify-content-end">
                  <Button primary
                    >Submit <Icon name="paperplane" class="ms-1"
                  /></Button>
                </div>
              </form>
              <hr />
              <div class="contact__details py-3">
                <p v-if="contact.location">
                  Location<Icon class="ms-1" name="location" />:<span
                    class="ms-2"
                    >{{ contact.location }}</span
                  >
                </p>
                <p v-if="contact.email">
                  Email<Icon class="ms-1" name="email" />:<a
                    class="ms-2"
                    :href="`mailto:${contact.email}`"
                    >{{ contact.email }}</a
                  >
                </p>
                <p v-if="contact.phone">
                  Phone<Icon class="ms-1" name="phone" />:<a
                    class="ms-2"
                    :href="`tel:${contact.phone}`"
                    >{{ contact.phone }}</a
                  >
                </p>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
    <Footer />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  async asyncData({ $content }) {
    const navbar = await $content('navbar').fetch()
    const contact = await $content('contact').fetch()
    return {
      navbar,
      contact,
    }
  },
  head: {
    title: 'Contact - Codefolio',
  },
})
</script>

<style lang="scss" scoped>
@import '@/scss/abstracts';
.contact {
  padding: pxToRem(10);
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: space-between;

  &__details > p {
    &:last-child {
      margin: 0;
    }
    display: flex;
    align-items: center;
    justify-content: flex-start;
    color: $gray-8;
    span {
      color: $primary;
    }
  }

  &__form {
  }

  &__label {
    color: $gray-6;
    font-size: pxToRem(15);
  }
  &__input {
    min-height: pxToRem(45);
  }
  &__input,
  &__textarea {
    transition: border 0.3s ease-in-out, box-shadow 0.3s ease-in-out;
    border: 1px solid rgba($gray-2, 0.7);
    @include hocus {
      border: 1px solid rgba($primary, 0.3);
      box-shadow: 0 0 pxToRem(5) 0 rgba($primary, 0.2);
    }
  }
}
</style>
