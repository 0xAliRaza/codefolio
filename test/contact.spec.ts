import { shallowMount } from '@vue/test-utils'
import contact from '@/pages/contact.vue'

describe('contact', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(contact, {
      data() {
        return {
          navbar: { resumeUrl: '', externalLinks: [] },

          contact: {
            formAction: '#',
            location: 'Example, USA.',
            email: 'contact@example.com',
            phone: '+1-555-555-0100',
          },
        }
      },
      stubs: ['Icon'],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
