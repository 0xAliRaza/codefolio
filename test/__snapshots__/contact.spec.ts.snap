// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact renders correctly 1`] = `
<div>
  <header resume-url
          external-links
  >
  </header>
  <main class="main">
    <div class="container">
      <div class="row justify-content-center">
        <div class="col-12 col-md-10 col-lg-8 col-xl-7 col-xxl-6">
          <section class="contact">
            <div class="contact__heading text-center my-3">
              <h1>
                Contact
              </h1>
            </div>
            <form action="#"
                  method="POST"
                  class="contact__form py-3"
            >
              <div class="mb-3">
                <label for="fullname"
                       class="form-label contact__label"
                >
                  Full name
                </label>
                <input id="fullname"
                       type="text"
                       name="fullname"
                       required="required"
                       class="form-control contact__input"
                >
              </div>
              <div class="mb-3">
                <label for="email"
                       class="form-label contact__label"
                >
                  Email
                </label>
                <input id="email"
                       type="email"
                       name="email"
                       required="required"
                       class="form-control contact__input"
                >
              </div>
              <div class="mb-3">
                <label for="subject"
                       class="form-label contact__label"
                >
                  Subject
                </label>
                <input id="subject"
                       type="text"
                       name="subject"
                       required="required"
                       class="form-control contact__input"
                >
              </div>
              <div class="mb-3">
                <label for="message"
                       class="form-label contact__label"
                >
                  Message
                </label>
                <textarea id="message"
                          name="message"
                          rows="4"
                          required="required"
                          class="form-control contact__textarea"
                >
                </textarea>
              </div>
              <div class="d-flex justify-content-end">
                <button primary>
                  Submit
                  <icon-stub name="paperplane"
                             class="ms-1"
                  >
                  </icon-stub>
                </button>
              </div>
            </form>
            <hr>
            <div class="contact__details py-3">
              <p>
                Location
                <icon-stub name="location"
                           class="ms-1"
                >
                </icon-stub>
                :
                <span class="ms-2">
                  Example, USA.
                </span>
              </p>
              <p>
                Email
                <icon-stub name="email"
                           class="ms-1"
                >
                </icon-stub>
                :
                <a href="mailto:contact@example.com"
                   class="ms-2"
                >
                  contact@example.com
                </a>
              </p>
              <p>
                Phone
                <icon-stub name="phone"
                           class="ms-1"
                >
                </icon-stub>
                :
                <a href="tel:+1-555-555-0100"
                   class="ms-2"
                >
                  +1-555-555-0100
                </a>
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
  <footer>
  </footer>
</div>
`;
